packages/melody-parser/src/Parser.js

```diff
@@ -289,6 +289,11 @@ export default class Parser {
                         expr.loc.start.column = column;
                         copyEnd(expr, expr.right);
                     }
+                    // Distinguish between BinaryConcatExpression generated by
+                    // this Parser, and those that the user wrote.
+                    if (nodes.length > 1) {
+                        expr.isMelodyGenerated = true;
```

Comment:
It'd be great if you could adapt the relevant class to include this field with a default value of `false`. That should help a lot with performance (regular, predictable class shape & monomorphic calls).

I'm also not 100% convinced about the name (`isMelodyGenerated`) but I can't come up with something better either. :)

Reply:
How do you like `wasImplicitConcatenation`?